src/frontend/js/data/useCourseEnrollment/index.ts

```diff
@@ -21,42 +21,45 @@ const useCourseEnrollment = (resourceLink: string) => {
   const queryClient = useQueryClient();
   const EnrollmentAPI = EnrollmentApiInterface(resourceLink);
 
-  const [{ data: enrollment, isError, isLoading }, enrollmentKeys] = useSessionQuery(
+  const [{ data: enrollment, isError, isLoading }, queryKeys] = useSessionQuery(
     ['enrollment', resourceLink],
-    async () => EnrollmentAPI.get(resourceLink, user!),
+    async () => {
+      return EnrollmentAPI.get(resourceLink, user!);
+    },
   );
 
   const [{ data: isActive, refetch: refetchIsActive }] = useSessionQuery(
-    [...enrollmentKeys, 'is_active'],
+    [...queryKeys, 'is_active'],
     async () => EnrollmentAPI.isEnrolled(enrollment),
     {
       // Enrollment is null if it has been fetched
       enabled: !!user && enrollment !== undefined && !isLoading,
     },
   );
 
-  useEffect(() => {
-    refetchIsActive();
-  }, [enrollment]);
-
   const { mutateAsync } = useSessionMutation(
     (activeEnrollment: boolean = true) =>
       EnrollmentAPI.set(resourceLink, user!, enrollment, activeEnrollment),
     {
-      mutationKey: enrollmentKeys,
+      mutationKey: queryKeys,
       onSuccess: async () => {
-        await queryClient.invalidateQueries(enrollmentKeys, { exact: true });
+        await queryClient.invalidateQueries(queryKeys, { exact: true });
         // After enrolls the user, then send enrolled event to the web analytics handler.
         WebAnalyticsAPIHandler()?.sendEnrolledEvent(resourceLink);
       },
     },
   );
 
+  useEffect(() => {
+    refetchIsActive();
+  }, [enrollment]);
+
   return {
     enrollment: isError ? null : enrollment,
     enrollmentIsActive: isActive,
     setEnrollment: mutateAsync,
     canUnroll: EnrollmentAPI.meta?.canUnroll ?? false,
+    isError,
```

Comment:
To homogenize with other hooks. What do you think to put error state is a states property ?

```ts
...
states: {
  errors: {
    get: isError,
  }
},
```

Reply:
much better